CMakeLists.txt

```diff
@@ -33,10 +33,10 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
 endif()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
@magnesj Ah thanks, you are absolutely right about the AppleClang Version! Meaning that this line is overdelivering: It should be `STREQUAL` instead of `MATCHES`, otherwise it wrongly matches with `AppleClang`.